tests/test_set.py

```diff
@@ -73,6 +74,147 @@ def test_text_metadata_multiple_chapters_ms(get_mp3):
     assert expected == actual.decode("utf-8")
 
 
+def test_text_metadata_apic_url(get_mp3):
+    expected = textwrap.dedent(
+        """\
+        TDRC = 2022-10-14
+        TRCK = 12/12
+        TPE1 = Ігор, Роман
+        TALB = Шо по коду?
+        TIT2 = Обробка помилок
+        TCON = Podcast
+        APIC = https://foo.bar/some.png
+
+        00:00:00 Кінець
+        """
+    )
+
+    test_mp3 = get_mp3("metadata.mp3")
+    subprocess.check_output(
+        ["id3manager", "set", test_mp3], input=expected.encode("utf-8")
+    )
+
+    actual = subprocess.check_output(["id3manager", "get", test_mp3])
+    assert expected == actual.decode("utf-8")
+
+
+def test_text_metadata_apic_file(get_mp3, testdata):
+    image_path = testdata / "logo.png"
+    image_data = open(image_path, "rb").read()
```

Comment:
nit: It seems you're using `pathlib` in which case I'd suggest embracing `pathlib` at full and do `image_data = image_path.read_bytes()`.

Reply:
I didn't know it exists :)